salt/utils/win_dotnet.py

```diff
@@ -0,0 +1,157 @@
+    for ver_key in ver_keys:
+
+        if ver_key.startswith('v'):
+            if win_reg.value_exists(hive=hive,
+                                    key='\\'.join([key, ver_key]),
+                                    vname='Version'):
+                # https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#find-net-framework-versions-1-4-with-codep
+                install = win_reg.read_value(
+                    hive=hive,
+                    key='\\'.join([key, ver_key]),
+                    vname='Install')['vdata']
+                if not install:
+                    continue
+                version = win_reg.read_value(
+                    hive=hive,
+                    key='\\'.join([key, ver_key]),
+                    vname='Version')['vdata']
+                sp = win_reg.read_value(
+                    hive=hive,
+                    key='\\'.join([key, ver_key]),
+                    vname='SP')['vdata']
+            elif win_reg.value_exists(
+                    hive=hive,
+                    key='\\'.join([key, ver_key, 'Full']),
+                    vname='Release'):
+                # https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#find-net-framework-versions-45-and-later-with-code
+                install = win_reg.read_value(
+                    hive=hive,
+                    key='\\'.join([key, ver_key, 'Full']),
+                    vname='Install')['vdata']
+                if not install:
+                    continue
+                version = dotnet_45_plus_versions(
+                    win_reg.read_value(
+                        hive=hive,
+                        key='\\'.join([key, ver_key, 'Full']),
+                        vname='Release')['vdata'])
+                sp = 'N/A'
+            else:
+                continue
+
+            service_pack = ' SP{0}'.format(sp) if sp != 'N/A' else ''
+            return_dict['versions'].append(version)
+            return_dict['details'][ver_key] = {
+                'version': version,
+                'service_pack': sp,
+                'full': '{0}{1}'.format(version, service_pack)}
+
+    return return_dict
+
+
+def versions_list():
+    '''
+    Get a sorted list of .NET versions installed on the system
+
+    Returns:
+        list: A sorted list of versions installed on the system
+    '''
+    return sorted(versions()['versions'])
+
+
+def versions_details():
+    '''
+    Get the details for all versions of .NET installed on a system
+
+    Returns:
+        dict: A dictionary of details for each version on the system. Contains
+        the following keys:
+            - version: The version installed
+            - service_pack: The service pack for the version installed
+            - full: The full version name including the service pack
+    '''
+    return versions()['details']
+
+
+def version_atleast(version):
```

Comment:
This seems like it should be `version_at_least` to be consistent with breaking between words.